### DISCUSSION TITLE
Non-existing registers

### DISCUSSION THREAD

**Discussion by Haismd03** [sandeepmistry/arduino-LoRa#629] [Q&A] [answered]
Hi, while working on a project I've taken closer look to how this library work.
I've compared register addresses with RFM96W datasheet and found out something I dont understand.

Register address is set to `0x39`
`#define REG_SYNC_WORD 0x39`
The `setSyncWord()` uses `writeRegister` that does bitwise OR with `0x80` and then a actually writes to resulted address `0xB9`.
This address is actally way beyond last register of RFM96W.

In datasheet, sync word register is only available in FSK/OOK mode. But this library works with LoRaTM. But yet, somehow it does the thing.

Can somebody tell me, how this library works?

**Comment by morganrallen** [comment, marked as answer]
Not sure where you're getting some of this info from but SyncWord in fact only works in LoRa mode, not FSK.
![image](https://user-images.githubusercontent.com/22881/218532455-e9e398d8-5153-4c1e-999a-7bbe335b75bc.png)


`0x80` in this case is the Write Enable Bit, actual register addresses are only 7-bit. This is _briefly_ described in section `4.3 SPI Interface`
```
The first byte is the address byte. It is comprises:
* A wnr bit, which is 1 for write access and 0 for read access.
* Then 7 bits of address, MSB first.
```

Hope this clears things up.

**Reply by Haismd03** [reply]
Thank you @morganrallen for clarification.
It's actually quite interesting, because it seem that I've been using wrong datasheet all the time.
In shop, where I've bought my RFM96W is pinned this [datasheet](https://www.laskakit.cz/user/related_files/rfm95_96_97_98w.pdf). It's clearly an older version, it's marked as v1.0.

![obrazek](https://user-images.githubusercontent.com/57405415/218544015-a77764fb-42ea-4efa-a367-4207926697ca.png)
In this version of datasheet, SyncWordConfig register has address `0x27` for FSK/OOK and is actually reserved for LoRa mode.

This is the reason why I was wrong about it. I'll notify admins of this e-shop about this issue.

Also thank you for notifiing me on section `4.3 SPI Interface`. I'll take closer look on it.
